docs/apib/compliancePolicies.apib

```diff
@@ -0,0 +1,214 @@
+FORMAT: 1A
+
+# XClarity compliancePolicies
+
+# get details of all policies
+
+## GET /compliancePolicies
++ Response 200
+{
+   "policies":[{
+      "description":"DEV ONLY",
+      "id":"1427930734934",
+      "inUse":"true",
+      "filterType":"all",
+      "lastModified":"2015-04-01 16:25:34",
+      "name":"DEFAULT-2015-04-01",
+      "updateRule":"",
+      "userDefined":"no"
+   },
+   {"updateRule":"",
+      "description":"ddddd",
+      "id":"1428397839104",
+      "inUse":"false",
+      "filterType":"managed",
+      "lastModified":"This policy was created by USERID on 2015-04-07 05:10:39",
+      "name":"aaaa",
+      "updateRule":"",
+      "userDefined":"yes"
+   }]
+}
++ Response 400
++ Response 401
++ Response 404
++ Response 409
++ Response 500
+
+# retrieve the persisted compare result for servers to which a compliance policy is assigned.
+
+## GET /compliancePolicies/persistedResult
++ Response 200
+{
+   "all": [{
+      "chassis": "FBEF740B178F4EFAA846E7225EE256DC",
+      "cmms": [{
+         "deviceCompliant": "no",
+         "message": [],
+         "policyName": "",
+         "supported": "true",
+         "targetFirmware": [],
+         "uuid": "387E51D69EC311E0A4C8E87E4D6C0479"
+      }]
```

Comment:
A comma is missing here

Reply:
its not required..its last member

Reply:
When I ran the tests via `bundle exec rspec spec` I got this:

```
Failures:

  1) XClarityClient GET /persistedResult with response 200 should not be nil
     Failure/Error: body = JSON.parse(response.body)
     
     JSON::ParserError:
       419: unexpected token at '{
         "chassis": "FBEF740B178F4EFAA846E7225EE256DC",
         "cmms": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "387E51D69EC311E0A4C8E87E4D6C0479"
         }]
         "switches": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "DEFAULT-2015-04-25",
            "supported": "true",
            "targetFirmware": [{
               "build": "0802",
               "category": "Switches",
               "compliant": "no",
               "componentID": "1B33D6C57ECB04C14567A897DC604900",
               "componentName": "IO Module 02",
               "date": "2015-04-22",
               "fixid": "lnvgy_fw_scsw_si4093-8.2.1.0_anyos_noarch",
               "level": "higher",
               "packageExistence": "yes",
               "reason": "The installed version is down level than the version defined in compliance policy."
               "versionList": ["8.2.1.0"]
            }],
            "uuid": "1B33D6C57ECB04C14567A897DC604900"
         }],
         "xITEs": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "69BDF8912E5211E4998B40F2E99033F0"
         }],
       "message": []
       },
       {
         "racklist": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "C29379AA380E11E39DF3000AF7256714"
         }]
       },
       {
         "rackswitchlist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "242CFF20453641238D2B8FEDFB0F056F"
         }]
       },
       {
         "storagelist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "208000C0FF25E476"
         }]
       }]
       }
     
       '
     # ./lib/xclarity_client/services/mixins/response_builder_mixin.rb:24:in `build_response_with_resource_list'
     # ./lib/xclarity_client/services/xclarity_service.rb:30:in `fetch_all'
     # ./lib/xclarity_client/services/persisted_result_management.rb:8:in `population'
     # ./lib/xclarity_client/client.rb:299:in `fetch_compliance_policies'
     # ./spec/xclarity_client_persisted_result_spec.rb:17:in `block (2 levels) in <top (required)>'
     # ./spec/xclarity_client_persisted_result_spec.rb:33:in `block (3 levels) in <top (required)>'

  2) XClarityClient GET /persistedResult with response 200 should has the xITEs
     Failure/Error: body = JSON.parse(response.body)
     
     JSON::ParserError:
       419: unexpected token at '{
         "chassis": "FBEF740B178F4EFAA846E7225EE256DC",
         "cmms": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "387E51D69EC311E0A4C8E87E4D6C0479"
         }]
         "switches": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "DEFAULT-2015-04-25",
            "supported": "true",
            "targetFirmware": [{
               "build": "0802",
               "category": "Switches",
               "compliant": "no",
               "componentID": "1B33D6C57ECB04C14567A897DC604900",
               "componentName": "IO Module 02",
               "date": "2015-04-22",
               "fixid": "lnvgy_fw_scsw_si4093-8.2.1.0_anyos_noarch",
               "level": "higher",
               "packageExistence": "yes",
               "reason": "The installed version is down level than the version defined in compliance policy."
               "versionList": ["8.2.1.0"]
            }],
            "uuid": "1B33D6C57ECB04C14567A897DC604900"
         }],
         "xITEs": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "69BDF8912E5211E4998B40F2E99033F0"
         }],
       "message": []
       },
       {
         "racklist": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "C29379AA380E11E39DF3000AF7256714"
         }]
       },
       {
         "rackswitchlist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "242CFF20453641238D2B8FEDFB0F056F"
         }]
       },
       {
         "storagelist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "208000C0FF25E476"
         }]
       }]
       }
     
       '
     # ./lib/xclarity_client/services/mixins/response_builder_mixin.rb:24:in `build_response_with_resource_list'
     # ./lib/xclarity_client/services/xclarity_service.rb:30:in `fetch_all'
     # ./lib/xclarity_client/services/persisted_result_management.rb:8:in `population'
     # ./lib/xclarity_client/client.rb:299:in `fetch_compliance_policies'
     # ./spec/xclarity_client_persisted_result_spec.rb:17:in `block (2 levels) in <top (required)>'
     # ./spec/xclarity_client_persisted_result_spec.rb:37:in `block (3 levels) in <top (required)>'

  3) XClarityClient GET /compliancePolicies/persistedResult should respond with an array
     Failure/Error: body = JSON.parse(response.body)
     
     JSON::ParserError:
       419: unexpected token at '{
         "chassis": "FBEF740B178F4EFAA846E7225EE256DC",
         "cmms": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "387E51D69EC311E0A4C8E87E4D6C0479"
         }]
         "switches": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "DEFAULT-2015-04-25",
            "supported": "true",
            "targetFirmware": [{
               "build": "0802",
               "category": "Switches",
               "compliant": "no",
               "componentID": "1B33D6C57ECB04C14567A897DC604900",
               "componentName": "IO Module 02",
               "date": "2015-04-22",
               "fixid": "lnvgy_fw_scsw_si4093-8.2.1.0_anyos_noarch",
               "level": "higher",
               "packageExistence": "yes",
               "reason": "The installed version is down level than the version defined in compliance policy."
               "versionList": ["8.2.1.0"]
            }],
            "uuid": "1B33D6C57ECB04C14567A897DC604900"
         }],
         "xITEs": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "supported": "true",
            "targetFirmware": [],
            "uuid": "69BDF8912E5211E4998B40F2E99033F0"
         }],
       "message": []
       },
       {
         "racklist": [{
            "deviceCompliant": "no",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "C29379AA380E11E39DF3000AF7256714"
         }]
       },
       {
         "rackswitchlist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "242CFF20453641238D2B8FEDFB0F056F"
         }]
       },
       {
         "storagelist": [{
            "deviceCompliant": "",
            "message": [],
            "policyName": "",
            "targetFirmware": [],
            "uuid": "208000C0FF25E476"
         }]
       }]
       }
     
       '
     # ./lib/xclarity_client/services/mixins/response_builder_mixin.rb:24:in `build_response_with_resource_list'
     # ./lib/xclarity_client/services/update_policy_management.rb:22:in `get_update_policy'
     # ./lib/xclarity_client/services/update_policy_management.rb:12:in `get_persisted_compare_results'
     # ./lib/xclarity_client/client.rb:359:in `discover_persisted_compare_results'
     # ./spec/xclarity_client_update_policy_spec.rb:51:in `block (3 levels) in <top (required)>'
```

Reply:
And when I do the requested changes, the tests pass successfully